connmgr/connmanager.go

```diff
@@ -466,7 +461,7 @@ func (cm *ConnManager) newConnReq(ctx context.Context) {
 	}
 
 	c := &ConnReq{}
-	atomic.StoreUint64(&c.id, atomic.AddUint64(&cm.connReqCount, 1))
```

Comment:
I don't see the point of using atomic access to `c.id` at this point since it's freshly constructed on the previous line, meaning there are no other references to this instance.   e.g. it could be

```go
c := &ConnReq{id: atomic.AddUint64(&cm.connReqCount, 1)}
```

Reply:
Agreed.  I'm doing some updates in a separate PR to address this and some other things.

Reply:
Done in #3055.